client/state/selectors/get-current-plan-purchase-id.js

```diff
@@ -0,0 +1,33 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { get, isNaN } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getCurrentPlan } from 'state/sites/plans/selectors';
+
+/**
+ * Returns a site's current plan purchase ID or null if the site doesn't exist or the purchase ID
+ * is unknown.
+ *
+ * The ID returned by this selector is retrieved from the site plan. It is intended to provide a
+ * means of retrieving full purhcase information based on a site and it's plan information.
+ * Caution! It _does not_ retrieve the ID from a purchase.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @return {?Number}        Purchase ID if known
+ */
+export default function getCurrentPlanPurchaseId( state, siteId ) {
```

Comment:
I'm pretty indifferent. I'm essentially adding a suffix to [`getCurrentPlan`](https://github.com/Automattic/wp-calypso/blob/887cd5b4e094f7c235de8cbd833604a9d268ef60/client/state/sites/plans/selectors.js#L36). The name isn't great as-is, but I'm not convinced adding `site` is an improvement. If you are, I'm happy to make the change.